### ISSUE TITLE
IPC Fatal Error: Leftover files in the directory

### ISSUE THREAD

**Issue by eserte** [fujiwara/Test-UNIXSock#3]
Every now and then t/15_oo_unix.t fails like this:
```
IPC Fatal Error: Leftover files in the directory (/var/tmp/cpansmoker-1023/2017092200/test2~19974~Bf5pU1/19974~0~1~19976~0~13~Test2~Event~Note.ready)!
Could not remove IPC temp dir (/var/tmp/cpansmoker-1023/2017092200/test2~19974~Bf5pU1) at /opt/perl-5.27.4/lib/site_perl/5.27.4/Test2/IPC/Driver/Files.pm line 357 during global destruction.
t/15_oo_unix.t ...... 
All 22 subtests passed 
```

A sample report at CPAN Testers: http://www.cpantesters.org/cpan/report/e02d57c4-9f98-11e7-bbca-cfbe104e4090

I think these failures started some time ago, probably with the introduction of Test2 (@exodist: FYI).

**Comment by ilmari** [comment]
I suspect the error happens when the parent process terminates before the child process, so the child's IPC files are still in the directory. Adding `undef $server;` (or `$server->stop();`) before the `done_testing();` call in the test  fixes it.

**Comment by clayne** [comment]
@ilmari I notice that change is still outstanding in another branch. Was it going to be merged in to master at some point?

https://github.com/fujiwara/Test-UNIXSock/compare/master...ilmari:patch-1

**Comment by ilmari** [comment]
@clayne I am not the maintainer, it's @fujiwara who needs to merge #4

**Comment by fujiwara** [comment]
@clayne @ilmari Sorry for late response. I've merged #4 and released v0.3 🙏
